### PR TITLE
Ensure ActiveIssue attribute is only used with issue links

### PR DESCRIPTION
**Pull request by akoeplinger**
The constructor that takes an `int` was removed in https://github.com/dotnet/arcade/pull/4831.

Update docs to recommend using issue links instead of numbers.

Fix two tests that were using ActiveIssue incorrectly.

### CODE REVIEW

File: src/libraries/System.Collections.Immutable/tests/ImmutableSortedSetTest.nonnetstandard.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ namespace System.Collections.Immutable.Tests
     public partial class ImmutableSortedSetTest : ImmutableSetTest
     {
         [Fact]
-        [ActiveIssue("Sporadic failure, needs a port of https://github.com/dotnet/coreclr/pull/4340", TargetFrameworkMonikers.NetFramework)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Sporadic failure, needs a port of https://github.com/dotnet/coreclr/pull/4340")]
```
Comment (akoeplinger) [inline src/libraries/System.Collections.Immutable/tests/ImmutableSortedSetTest.nonnetstandard.cs]: I thought about filing an issue for this but given that https://github.com/dotnet/coreclr/pull/4340 is four years old now and wasn't ported to .NET Framework we can probably assume it won't happen.

File: src/libraries/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs [hunk 2]
```diff
@@ -3493,8 +3493,7 @@ public static void DCS_BasicPerSerializerRoundTripAndCompare_SampleTypes_SampleI
     /// <summary>
     /// This case is a part of DCS_BasicPerSerializerRoundTripAndCompare_SampleTypes, but in these scenarios it was not support in current version.
     /// </summary>
-    [Fact]
-    [ActiveIssue("Not support")]
+    [Fact(Skip = "Not supported")]
```
Comment (stephentoub) [inline src/libraries/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: If we're going to always skip the test and there's no issue / no follow-up, shouldn't we just delete the test / dead code?

Comment (akoeplinger) [reply src/libraries/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: Sure, whatever you prefer.

Comment (stephentoub) [reply src/libraries/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: Thanks. Let's delete it.

Comment (akoeplinger) [reply src/libraries/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: Done

**Review by stephentoub — APPROVED** [review]
Thanks for cleaning this up.

**Review by ViktorHofer — COMMENTED** [review]
Thanks, lgtm

**Review by safern — COMMENTED** [review]
LGTM

**Pull request merged**